src/Adapter/Presenter/Product/ProductLazyArray.php

```diff
@@ -1138,17 +1138,23 @@ public function addQuantityInformation(
             $this->product['availability_date'] = $product['available_date'];
             $this->product['availability'] = 'unavailable';
 
+            // We will primarily use label from combination if set, then label on product, then the default label from PS settings
+            if (!empty($combinationData['available_later'])) {
+                $this->product['availability_message'] = $combinationData['available_later'];
+            } elseif (!empty($product['available_later'])) {
+                $this->product['availability_message'] = $product['available_later'];
+            } else {
+                $config = $this->configuration->get('PS_LABEL_OOS_PRODUCTS_BOD');
+                $this->product['availability_message'] = $config[$language->id] ?? null;
+            }
+
             // If the product has combinations and other combination is in stock, we show a small hint about it
             if ($product['cache_default_attribute'] && $product['quantity_all_versions'] > 0) {
-                $this->product['availability_message'] = $this->translator->trans(
+                $this->product['availability_submessage'] = $this->translator->trans(
```

Comment:
so, hummingbird PR to follow, yes?

Reply:
https://github.com/PrestaShop/hummingbird/pull/658